src/charm.py

```diff
@@ -7,48 +7,46 @@
 Upstream doc: https://argoproj.github.io/argo-rollouts/
 """
 
+from glob import glob
 import logging
 import re
 import requests
-import traceback
+import time
 
-from charms.grafana_k8s.v0.grafana_dashboard import GrafanaDashboardProvider
-from charms.loki_k8s.v0.loki_push_api import LogProxyConsumer
-from charms.prometheus_k8s.v0.prometheus_scrape import MetricsEndpointProvider
-
-from glob import glob
 from lightkube import Client, codecs
 from lightkube.core.exceptions import ApiError
-from ops.charm import CharmBase, PebbleReadyEvent
-from ops.main import main
-from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
-from ops.pebble import LayerDict, Layer
+import ops
 
+from charms.grafana_k8s.v0.grafana_dashboard import GrafanaDashboardProvider
+from charms.loki_k8s.v0.loki_push_api import LogProxyConsumer
+from charms.prometheus_k8s.v0.prometheus_scrape import MetricsEndpointProvider
 
 logger = logging.getLogger(__name__)
 
+METRICS_PORT = 8090
+MAX_PEBBLE_RETRIES = 3
+PEBBLE_RETRY_DELAY = 3
+
 
-class ArgoRolloutsOperatorCharm(CharmBase):
+class ArgoRolloutsCharm(ops.CharmBase):
     """Charmed Operator for Argo Rollouts."""
 
-    def __init__(self, *args):
-        super().__init__(*args)
+    def __init__(self, framework):
+        super().__init__(framework)
 
         self.pebble_service_name = "argo-rollouts"
         self.container = self.unit.get_container("argo-rollouts")
         self._context = {"namespace": self._namespace, "app_name": self.app.name}
 
-        self.framework.observe(
-            self.on.argo_rollouts_pebble_ready, self._argo_rollouts_pebble_ready
-        )
-        self.framework.observe(self.on.install, self._on_install_or_upgrade)
-        self.framework.observe(self.on.upgrade_charm, self._on_install_or_upgrade)
-        self.framework.observe(self.on.remove, self._on_remove)
+        framework.observe(self.on.argo_rollouts_pebble_ready, self._install_and_restarty)
```

Comment:
Is the "y" on the end a typo?
```suggestion
        framework.observe(self.on.argo_rollouts_pebble_ready, self._install_and_restart)
```